console/services/app.py

```diff
@@ -903,6 +903,27 @@ def create_market_app_model(self, market, body):
     def create_market_app_model_version(self, market, app_id, body):
         app_store.create_app_version(market, app_id, body)
 
+    def list_bindable_markets(self, eid, market_name, market_url, access_key):
+        if market_name:
+            market = app_market_service.get_app_market_by_name(eid, market_name)
+        else:
+            market = AppMarket(url=market_url, access_key=access_key)
+
+        bindable_markets = app_store.list_bindable_markets(market)
+        if not bindable_markets:
+            return None
+
+        # make up bound markets
+        bound_markets = set()
+        for bm in self.get_app_markets(eid, "true"):
+            if not (len(bm["access_actions"]) == 1 and bm["access_actions"][0] == "OnlyRead"):
+                continue
```

Comment:
Why filter?

Reply:
If the bound market is read-only, then this market should be in the bindable list. Rebinding is allowed.

Reply:
Exclude market that is already bound, It should only be related to the URL + Domain condition.

Reply:
The default store "Rainbond Open Source Store" in the user environment is read-only. When the user is authenticated and bound, if the read-only store is not listed, the default store cannot be modified.

Reply:
Is this a question of how to change authorization?

Reply:
Changing the authentication can be to delete the old one and add the new one again. Here it is just to adapt to the default market

Reply:
There is no problem with the market being bound multiple times. Different permissions can be bound. so, do not filter.

Reply:
ok